spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/JvmMetricsAutoConfiguration.java

```diff
@@ -53,10 +50,17 @@
 @AutoConfiguration(after = { MetricsAutoConfiguration.class, CompositeMeterRegistryAutoConfiguration.class })
 @ConditionalOnClass(MeterRegistry.class)
 @ConditionalOnBean(MeterRegistry.class)
-public class JvmMetricsAutoConfiguration {
+public class JvmMetricsAutoConfiguration implements BeanClassLoaderAware {
 
 	private static final String VIRTUAL_THREAD_METRICS_CLASS = "io.micrometer.java21.instrument.binder.jdk.VirtualThreadMetrics";
 
+	private ClassLoader beanClassLoader;
+
+	@Override
+	public void setBeanClassLoader(ClassLoader classLoader) {
+		this.beanClassLoader = classLoader;
+	}
+
```

Comment:
I think we should simplify this further and use the class loader that loaded `JvmMetricsAutoConfiguration`.  Using the bean class loader theoretically allows `JvmMetricsAutoConfiguration` to be loaded by one class loader and `VirtualThreadMetrics` to be loaded by another. I don't think we need to support that arrangement as it wouldn't be supported if this were a straightforward `@Bean` method with `VirtualThreadMetrics` as its return type.